src/Clerk.ts

```diff
@@ -276,6 +285,50 @@ export default class Clerk {
     });
   }
 
+  withProtectedRouteSSR(destination: string){
+    return (handler?: (ctxWithSession: SSRPropsContext) => Promise<GetServerSidePropsResult<any>>) => async (ctx: GetServerSidePropsContext) => {
+
+      // GetServerSidePropsContext req does not contain the query.
+      // For the sake of keeping the abstraction cleaner we inject it from ctx
+      // @ts-ignore
+      ctx.req.query = ctx.query;
+
+      try{
+        await this._runMiddleware(ctx.req, ctx.res, this.expressWithSession({ onError: this.strictOnError }));        
+      }catch(err){
```

Comment:
```suggestion
      } catch(err) {
```

Reply:
Sadly `prettier` configuration and runtime comes directly from `tsdx`. Because tsdx uses an outdated version of TS, prettier fails so it does not work on any file that has any "new" feature e.g. `import type...`